File: python/anagram/anagram.py
def isAnagram(w1, w2):
  if len(w1) != len(w2) or w1 == w2:
    return False
  for i in w1:
    if i not in w2:
      return False
    w2 = w2.replace(i, '', 1)
  return len(w2) == 0

def find_anagrams(word, candidates):
  return [c for c in candidates if isAnagram(word.lower(), c.lower())]

File: dart/anagram/lib/anagram.dart
class Anagram {

  bool _isAnagram(List word, String candidate) {
    if (candidate.length != word.length) {
      return false;
    }
    List<String> c = candidate.split('');
    c.sort();
    for (int i = 0; i < c.length; i++) {
      if (c[i] != word[i]) {
        return false;
      }
    }
    return true;
  }

  List<String> findAnagrams(String word, List<String> candidates) {
    List<String> res = [];
    String lowerWord = word.toLowerCase();
    List<String> processedWord = word.toLowerCase().split('');
    processedWord.sort();
    candidates.forEach((c) => lowerWord != c.toLowerCase() && _isAnagram(processedWord, c.toLowerCase()) ? res.add(c) : null);
    return res;
  }
}
